File: src/main/java/ar/edu/um/tesoreria/rest/model/Dependencia.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Dependencia extends Auditable implements Serializable {
	
	private static final long serialVersionUID = -7619347846778510021L;

	@Id
	private Integer dependenciaId;

	private String nombre = "";
	private String acronimo = "";
	private Integer facultadId;
	private Integer geograficaId;
	
	@OneToOne(optional = false)
	@JoinColumn(name = "facultadId", insertable = false, updatable = false)
	private Facultad facultad;

	@OneToOne(optional = false)
	@JoinColumn(name = "geograficaId", insertable = false, updatable = false)
	private Geografica geografica;

	public String getSedeKey() {
		return this.facultadId + "." + this.geograficaId;
	}

}

File: src/main/java/um/tesoreria/rest/kotlin/model/Dependencia.kt
package um.tesoreria.rest.kotlin.model

import jakarta.persistence.*

@Entity
@Table
data class Dependencia(

    @Id
    var dependenciaId: Int? = null,
    var nombre: String = "",
    var acronimo: String = "",
    var facultadId: Int? = null,
    var geograficaId: Int? = null,

    @OneToOne(optional = true)
    @JoinColumn(name = "facultadId", insertable = false, updatable = false)
    val facultad: Facultad? = null,

    @OneToOne(optional = true)
    @JoinColumn(name = "geograficaId", insertable = false, updatable = false)
    var geografica: Geografica? = null,


) : Auditable() {

    fun getSedeKey(): String {
        return this.facultadId.toString() + "." + this.geograficaId
    }

}
